app/views/users/_form.html.erb

```diff
@@ -0,0 +1,25 @@
+<%= form_tag path, method: method do %>
+  <section>
+    <%= label_tag "Name: " %>
+    <%= text_field_tag :name, user.name %><br>
+    <%= label_tag "Street Address: " %>
+    <%= text_field_tag :address, user.address %><br>
+    <%= label_tag "City: " %>
+    <%= text_field_tag :city, user.city %><br>
+    <%= label_tag "State: " %>
+    <%= text_field_tag :state, user.state %><br>
+    <%= label_tag "Zip Code: " %>
+    <%= text_field_tag :zip, user.zip %><br>
+    <%= label_tag "Email Address: " %>
+    <%= text_field_tag :email%><br>
+
+    <% if needs_password %>
+      <%= label_tag "Password: " %>
+      <%= password_field_tag :password %><br>
+      <%= label_tag "Confirm Password: " %>
+      <%= password_field_tag :password_confirmation %><br>
+    <% end %>
```

Comment:
Nice conditional, I'm not seeing it in this PR, but where is this method located, is it rails magic? ✨

Reply:
FOUND IT! Creative use of rendering partials and sending in a boolean along with the locals!